src/android/mm/generic_frame_buffer_allocator.cpp

```diff
@@ -128,9 +123,33 @@ PlatformFrameBufferAllocator::Private::allocate(int halPixelFormat,
 	std::vector<FrameBuffer::Plane> planes(info.numPlanes());
 
 	SharedFD fd{ handle->data[0] };
+	const size_t maxDmaLength = lseek(fd.get(), 0, SEEK_END);
+	
+	LOG(HAL, Debug) << "Private::allocate: created fd=" << fd.get() 
+					<< " pixelFormat=" << info.name 
+					<< " stride=" << stride
+					<< " numPlanes=" << info.numPlanes()
+					<< " numFds=" << handle->numFds
+					<< " numInts=" << handle->numInts
+					<< " dmaLength=" << maxDmaLength;
+
+	for(int i=0; i<handle->numFds; i++) {
+		int fdd = handle->data[i];
+		const size_t len = lseek(fdd, 0, SEEK_END);
+		LOG(HAL, Debug) << "Private::allocate: fd info fd=" << fdd << " len=" << len;
+	}
+	
 	size_t offset = 0;
 	for (auto [i, plane] : utils::enumerate(planes)) {
-		const size_t planeSize = info.planeSize(size.height, i, stride);
+		//SharedFD fdd{ handle->data[i] };
+		//size_t planeSize = info.planeSize(size.height, i, stride);
+		size_t planeSize = info.planeSize(size, i);
+
+		// if(planeSize > maxDmaLength) {
+		// 	planeSize = maxDmaLength;
+		// }
+
+		LOG(HAL, Debug) << "Private::allocate: planeInfo i=" << i << " offset=" << offset << " size=" << planeSize;
```

Comment:
This code, even if diverged from mainline substantially, seems correct.

It creates two logical planes of the right sizes 

```
07-09 12:54:51.528   308   308 D libcamera: DEBUG HAL generic_frame_buffer_allocator.cpp:152 Private::allocate: planeInfo i=0 offset=0 size=307200
07-09 12:54:51.528   308   308 D libcamera: DEBUG HAL generic_frame_buffer_allocator.cpp:152 Private::allocate: planeInfo i=1 offset=307200 size=153600
```

with a single `fd`